erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -231,15 +231,37 @@ frappe.ui.form.on('Stock Entry', {
 			}, __("Get Items From"));
 
 			frm.add_custom_button(__('Material Request'), function() {
-				erpnext.utils.map_current_doc({
+				const allowed_request_types = ["Material Transfer", "Material Issue", "Customer Provided"];
+				const depends_on_condition = "eval:doc.material_request_type==='Customer Provided'";
+				const d = erpnext.utils.map_current_doc({
 					method: "erpnext.stock.doctype.material_request.material_request.make_stock_entry",
 					source_doctype: "Material Request",
 					target: frm,
 					date_field: "schedule_date",
-					setters: {},
+					setters: [{
+						fieldtype: 'Select',
+						label: __('Purpose'),
+						options: allowed_request_types.join("\n"),
+						fieldname: 'material_request_type',
+						default: "Material Transfer",
+						mandatory: 1,
```

Comment:
The use case mentioned makes sense, but let's not make this mandatory due to the volume of edge cases we see often.
Usually the user raising the material request will differ from the one raising the stock entry.
MR user could want to **Issue** some stock, but SE user may decide to **transfer** that _Issued_ stock elsewhere. So its hard to tell what MR purpose, maps to what Stock Entry type.